app/controllers/users_controller.rb

```diff
@@ -0,0 +1,48 @@
+class UsersController < ApplicationController
+  get '/signup' do
+    if session[:id].nil?
+      erb :'users/signup'
+    else
+      redirect '/tweets'
+    end
+  end
+
+  post '/signup' do
+    if (params[:username].empty? || params[:password].empty? || params[:email].empty?)
+      redirect '/signup'
+    else
+      user = User.create(params)
+      session[:id] = user.id
+      redirect '/tweets'
+    end
+  end
+
+  get '/login' do
+    if !session[:id].nil?
+      redirect '/tweets'
+    else
+      erb :'users/login'
+    end
+  end
+
+  post '/login' do
+    user = User.find_by(username: params[:username])
+    session[:id] = user.id
+    if user.authenticate(params[:password])
```

Comment:
we should check to make sure a user exists as well as authenticate the password. we should also set session[:user_id] once we authenticate too.

```ruby 
if user && user.authenticate(params[:password]
  session[:user_id] = user.id 
  redirect '/tweets'
```